src/diffusers/utils/loading_utils.py

```diff
@@ -47,3 +50,77 @@ def load_image(
         image = image.convert("RGB")
 
     return image
+
+
+def load_video(
+    video: Union[str, List[PIL.Image.Image]],
+    convert_method: Optional[Callable[[List[PIL.Image.Image]], List[PIL.Image.Image]]] = None,
+) -> List[PIL.Image.Image]:
+    """
+    Loads `video` to a list of PIL Image.
+
+    Args:
+        video (`str` or `List[PIL.Image.Image]`):
+            The video to convert to a list of PIL Image format.
+        convert_method (Callable[[List[PIL.Image.Image]], List[PIL.Image.Image]], *optional*):
+            A conversion method to apply to the video after loading it. When set to `None` the images will be converted
+            to "RGB".
+
+    Returns:
+        `List[PIL.Image.Image]`:
+            The video as a list of PIL images.
+    """
+    if isinstance(video, str):
+        was_tempfile_created = False
+
+        if video.startswith("http://") or video.startswith("https://"):
+            video_data = requests.get(video, stream=True).raw
+            video_path = tempfile.NamedTemporaryFile(suffix=os.path.splitext(video)[1], delete=False).name
+            was_tempfile_created = True
+            with open(video_path, "wb") as f:
+                f.write(video_data.read())
+            video = video_path
+        elif not os.path.isfile(video):
+            raise ValueError(
+                f"Incorrect path or URL. URLs must start with `http://` or `https://`, and {video} is not a valid path."
+            )
+
+        if video.endswith(".gif"):
+            pil_images = []
+            gif = PIL.Image.open(video)
+            try:
+                while True:
+                    pil_images.append(gif.copy())
+                    gif.seek(gif.tell() + 1)
+            except EOFError:
+                pass
+        else:
+            if is_opencv_available():
+                import cv2
+            else:
+                raise ImportError(BACKENDS_MAPPING["opencv"][1].format("load_video"))
+            pil_images = []
+            video_capture = cv2.VideoCapture(video)
+            success, frame = video_capture.read()
+            while success:
+                frame = cv2.cvtColor(frame, cv2.COLOR_BGR2RGB)
+                pil_images.append(PIL.Image.fromarray(frame))
+                success, frame = video_capture.read()
+            video_capture.release()
+
+        if was_tempfile_created:
+            os.remove(video_path)
+
+    elif isinstance(video, list) and all(isinstance(frame, PIL.Image.Image) for frame in video):
```

Comment:
If passing a list of PIL images, we would just return the same list back? Why support passing in a list of images then?

Reply:
I followed the implementation of [load_image](https://github.com/huggingface/diffusers/blob/bce9105ac79636f68dcfdcfc9481b89533db65e5/src/diffusers/utils/loading_utils.py#L35). And the same list might not be returned since in the code that follows, `convert_method` ensures a callback is called or we convert the images to RGB (possibly from RGBA/HSV/etc.)

Reply:
Hmm seems like there's been a bit of back and forth about it
https://github.com/huggingface/diffusers/pull/6479
https://github.com/huggingface/diffusers/pull/6904

IMO it doesn't make much sense to pass a list of already loaded images into the `load_video` just to run preprocessing. If the user is at a point where they already have this list of images, it should be up to them to preprocess on their own. 

I think the conversion to RGB by default should also be removed. A loading function should just return the objects. If additional processing has to be done, it can be done via the `convert_method`

Reply:
I see, okay yeah that makes sense. Will remove this change